### ISSUE TITLE
Repeated key in switch clause produces unparseable code

### ISSUE THREAD

**Issue by gwhitney** [DanielXMoore/Civet#692]
Currently ( see [Playground](https://civet.dev/playground/?code=c3dpdGNoIGRhdGEgOj0ge2E6ICd6JywgYjogJ3knfQogICB7YTogJ3onLCBhOiAneCd9CiAgICAgIGNvbnNvbGUubG9nICdiaXphcnJlJw%3D%3D) )
```
switch data := {a: 'z', b: 'y'}
   {a: 'z', a: 'x'}
      console.log 'bizarre'
```
produces the following block of code for the case:
```
      const {a,: a1 a: a2} = m;
      const a = [a1, a2];
      console.log('bizarre')}
```
which is not parseable TypeScript. I hit this because of a typo and I'm not sure that this is a particularly productive pattern (I mean, you could test that a property satisfies both of two regular expressions, which otherwise would have to be expressed in a somewhat opaque way with lookahead assertions, but that seems like a pretty rare need). However, it should either error or produce parseable code I would think.

**Comment by gwhitney** [comment]
Ah, I see, it must be some sort of unintended side effect of
```
INFO

Patterns can aggregate duplicate bindings.

switch x
  [{type}, {type}]
    type
```
from the cheatsheet.

**Comment by STRd6** [comment]
Definitely a bug, thanks for the report!

**Comment by STRd6** [comment]
Released in v0.6.30
